lab/webapp/src/components/Dataset/components/DatasetMenu/components/Charts/BarChart/index.jsx

```diff
@@ -0,0 +1,172 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Header } from 'semantic-ui-react';
+import * as d3 from "d3";
+
+class BarChart extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+
+    };
+  this.createBarGraph = this.createBarGraph.bind(this);
+  }
+
+  componentDidMount() {
+    this.createBarGraph();
+  }
+
+  // adapted from https://bl.ocks.org/d3noob/bdf28027e0ce70bd132edc64f1dd7ea4
+  createBarGraph(){
+    const { dataPreview, valByRowObj, depCol, cleanKey } = this.props;
+    let margin = { top: 10, right: 30, bottom: 20, left: 40 },
+        width = 400 - margin.left - margin.right,
+        height = 200 - margin.top - margin.bottom;
+
+    let chartInnerHTML = "";
+    //let valByRowObj = this.getDataValByRow();
+    if(document.getElementById("test_bar_chart_" + depCol)) {
+      chartInnerHTML = document.getElementById("test_bar_chart_" + depCol).innerHTML;
+    };
+
+    if(chartInnerHTML === "") {
+      width = 460 - margin.left - margin.right;
+      let data_sorted = valByRowObj[depCol].sort(d3.ascending);
+
+      let classCountObj = {};
+      let chartData = [];
+
+      data_sorted.forEach(val => {
+        classCountObj[val] = classCountObj[val] ? ++classCountObj[val] : 1;
+      })
+      //chartData.push(classCountObj);
+      let testSet = [... new Set(valByRowObj[depCol])].sort();
+
+      /**---- *************** ----**** ---- Color stuff here ----****---- *************** ----**/
+      // for every entry in testSet, map keys to color
+      let colorObjList = [];
+      testSet.forEach((depVal, i) => {
+        // use https://github.com/d3/d3-scale-chromatic#schemePaired for 12 colors
+        // to select unique color per class in dataset
+        let colorString;
+        if(i < 12) {
+          colorString = d3.schemePaired[i];
+        } else {
+          // if more than 12 classes in dataset use sequential color range
```

Comment:
also note that if there are more than 12 unique classes then every unique class value color will be from a spectrum - I did this as a way to guarantee each class gets a distinct color (using a spectrum there are infinite colors but the contrast between entries can be small). 

I can also use distinct colors in a range like for the first twelve too